CDL-Dryad/dryad-app
# Fix additional file in Merritt when users change filename case across versions

When a user uploaded identical filenames, but with different case *across multiple versions* then the previous file might not be deleted correctly from Merritt (zip downloads).  This has no effect if a user uploaded different cases of the same filename and deleted them within the same version.  Also a side effect is that it causes weirdness extracting zip files on some file systems with different case names. Some of the filesystems consider different cases to be unique files and some consider them to be the same file (and overwrite an earlier file with a later one if they consider them to be the same).

To manually test:

1. Create a few files with the same name but different case and unique contents so they're easy to tell apart.
2. Submit a dataset with one of the files, wait for it to go through.
3. Copy the Merritt URL from the `stash_engine_resources.download_url` change `/d/` to `/m/` in the URL and open in your browser.  You'll see the files in Merritt.
4. Edit the same dataset again.
5. Attempt uploading the file with different case but the same name.  You should get a message indicating that the file already exists. (this only appeared before if the files had identical filenames with the case being the same.)
6. Remove the existing file and upload the new file with the different case.
7. Submit the dataset and wait for it go through.
8. Check Merritt for the new version to be sure it appears correctly and the old filename is gone.

If you like, you can also check the stash_engine_generic_files table if you wish for the 2nd resource_id you submitted and you should see that it's creating a `delete` action in Merritt for the previous filename which differs from current filename that was added.
